File: stackQueue/RightSideView199.java
public class RightSideView199{
    public static void main(String[] args) {

    }

    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> res=new ArrayList<>();
        if(root==null) return res;
        LinkedList<TreeNode> queue=new LinkedList<>();
        queue.add(root);
        while(!queue.isEmpty()){
            int count=queue.size();
            while(count>0){
                TreeNode node=queue.poll();
                if (node.left!=null) {
                    queue.add(node.left);
                }
                if (node.right!=null) {
                    queue.add(node.right);
                }
                if (count==1) {
                    res.add(node.val);
                }
                count--;
            }
        }
        return  res;
    }
}

File: tree/RightSideView199.go
func rightSideView(root *TreeNode) []int {
    var dfs func(root *TreeNode, depth int)
    var res = make([]int, 0)
    dfs = func(root *TreeNode, depth int) {
        if root == nil {
            return
        }
        if depth > len(res) {
            res = append(res, root.Val)
        }
        dfs(root.Right, depth+1)
        dfs(root.Left, depth+1)
    }
    dfs(root, 1)
    return res
}
